2-projects/1-todo-list/script.js

```diff
@@ -1,5 +1,57 @@
+// ## Instructions
+
+// The `populateTodoList()` function should iterate over the list of todos that we are given at the start, it should create a `<li>` for the todo along with some other stuff that you can find in index.html and below.
+
+// The items in the todo list are currently hard-coded into the HTML, refactor the code so that this function creates them and adds the following functionality to them:
+
+// Each todo should have this HTML inside it:
+
+// ```html
+// <span class="badge bg-primary rounded-pill">
+//   <i class="fa fa-check" aria-hidden="true"></i>
+//   <i class="fa fa-trash" aria-hidden="true"></i>
+// </span>
+// ```
+
+// The first `<i>` tag needs an event listener that applies a line-through text-decoration styling to the text of the todo. It should remove the styling if it is clicked again.
+
+// The second `<i>` tag needs an event listener that deletes the parent `<li>` element from the `<ul>`.
+let list = document.getElementById("todo-list");
+
+while (list.hasChildNodes()) {
+  list.removeChild(list.firstChild);
+}
 function populateTodoList(todos) {
-  let list = document.getElementById("todo-list");
+  
+  //let newList = document.createElement('ul');
+  //newList.setAttribute('id', 'todo-list');
+  //newList.setAttribute('class', 'list-group');
+
+    todos.forEach((item) => {
+    let listEl = document.createElement("li");
+    listEl.className =
+      "list-group-item d-flex justify-content-between align-items-center";
+    listEl.innerHTML = item.task;
+    let span = document.createElement("span");
+    span.className = "badge bg-primary rounded-pill";
+    let iElement = document.createElement("i");
+    iElement.className = "fa fa-check";
+    iElement.setAttribute("aria-hidden", "true");
+    let iElement2 = document.createElement("i");
+    span.appendChild(iElement);
+    iElement.addEventListener("click", () => {
```

Comment:
We could make use of iElement.onclick = function(){ #put your function here }

Reply:
Also when clicking the checkmark, it first crosses out the item (as expected) but then it deletes it. You may want to double check your event handlers!